### ISSUE TITLE
Configurable timeout?

### ISSUE THREAD

**Issue by jfeldstein** [DracoBlue/node-facebook-client#20]
Some of my queries sometimes take longer than 10 seconds to execute. 

I tried to fork and make a patch to pass in a custom timeout, but your function returns function way of structuring things makes it very hard to get down to the core components. 

Any way you could add a configurable timeout?

Thanks,
~Jordan

```
btw, this library rocks. Thanks. 
```

**Comment by jfeldstein** [comment]
For what it's worth, I'm also really curious why you structured this library in such a strange way. Is it just a style, or is it helpful in a way I don't quite understand? Thx for explaining

**Comment by DracoBlue** [comment]
Do you mean the continuous style: `return function(cb)`? Or the separation between FacebookClient+FacobookSession?

**Comment by jfeldstein** [comment]
The Return function (cb) pattern

~J

(via phone)
On Jan 11, 2012 12:58 AM, "DracoBlue" <
reply@reply.github.com>
wrote:

**Comment by DracoBlue** [comment]
The advantage is, that one can add optional parameters (even functions), without breaking the signature of the method:

```
#graphCall(method, params)(function() {
});
```

becomes to:

```
#graphCall(method, params, options)(function() {
     console.log('timeout is', options.timeout);
});
```

But I keep this pattern in this library just for consistence. It's pretty difficult to maintain and doesn't help autocompletion in IDE and such things :).

**Comment by jfeldstein** [comment]
Neato. 

So the best way to add this timeout parameter would be to just add it as an argument of the top-level method, and pass it all the way down to onRequest?

**Comment by leoasis** [comment]
+1 on this, it should not be hard to add a last options parameter with a configurable timeout.

**Comment by DracoBlue** [comment]
Fixed with 593c6f9c5c7149ad9a488f577f2893b743fdd3c0
